api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupTxCurrent_28Days2019ReportPatientListReport.java

```diff
@@ -0,0 +1,165 @@
+    @Autowired
+    private HIVCohortDefinitionLibrary hivCohortDefinitionLibrary;
+
+    @Autowired
+    private BuiltInPatientDataLibrary builtInPatientData;
+
+    @Autowired
+    private HIVPatientDataLibrary hivPatientData;
+
+    @Autowired
+    private BasePatientDataLibrary basePatientData;
+
+    /**
+     * @return the uuid for the report design for exporting to Excel
+     */
+    @Override
+    public String getExcelDesignUuid() {return "37e50e55-1cef-4f80-933d-006f49a3141f";}
+
+    public String getCSVDesignUuid() {
+        return "f6e7fe18-b5e2-470b-be29-f3d883f56feb";
+    }
+
+    @Override
+    public String getUuid() {
+        return "28cc02a3-800d-4638-bfb2-a56971355502";
+    }
+
+    @Override
+    public String getName() {
+        return "Patient List for Tx Current 28 days MER Indicator Report";
+    }
+
+    @Override
+    public String getDescription() {
+        return "TX_Current 28 days Patient List Report";
+    }
+
+    @Override
+    public List<Parameter> getParameters() {
+        List<Parameter> l = new ArrayList<>();
+        l.add(df.getStartDateParameter());
+        l.add(df.getEndDateParameter());
+        return l;
+    }
+
+
+    @Override
+    public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+        List<ReportDesign> l = new ArrayList<>();
+        l.add(buildReportDesign(reportDefinition));
+        l.add(buildCSVReportDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding
+     * properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+    @Override
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "MER_TX_CUR_PatientList.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:8,dataset:TX_CUR_PatientList");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    public ReportDesign buildCSVReportDesign(ReportDefinition reportDefinition) {
+        return createCSVDesign(getCSVDesignUuid(), reportDefinition);
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.setParameters(getParameters());
+
+        PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+
+        dsd.setParameters(getParameters());
+        rd.addDataSetDefinition("TX_CUR_PatientList", Mapped.mapStraightThrough(dsd));
+
+        CohortDefinition returnToCareClients= hivCohortDefinitionLibrary.getActivePatientsWithLostToFollowUpAsByDays("28");
```

Comment:
Rename the cohort definition to a proper name according to the report

Reply:
@dbaluku Done